### ISSUE TITLE
SeqIO.parse: guess format from file extension, if present

### ISSUE THREAD

**Issue by anntzer** [biopython/biopython#758]
All's in the title: when a filename is passed to SeqIO.parse (rather than a handle), the format could be guessed from the extension, if present.

**Comment by mdehoon** [comment]
The alternative is to have format='fasta' by default.

**Comment by anntzer** [comment]
I'd say if I pass a .gb file it is clear that it should be genbank (extensions are there for a reason...).

**Comment by mdehoon** [comment]
See the previous discussion here:
http://lists.open-bio.org/pipermail/biopython/2011-October/013761.html
and here:
http://lists.open-bio.org/pipermail/biopython-dev/2006-November/011666.html
That said, if you can convince enough users & developers that this is a good idea, then I won't stop you.

**Comment by anntzer** [comment]
I agree 100% that guessing the format from the contents is a terrible idea (thus the exclusion of handles from the proposal).  I am only proposing a strict extension -> format table, as a convenience for the user.
